flink-runtime/src/test/java/org/apache/flink/runtime/concurrent/ManuallyTriggeredComponentMainThreadExecutor.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.concurrent;
+
+/** Testing ComponentMainThreadExecutor which can be manually triggered. */
+public class ManuallyTriggeredComponentMainThreadExecutor
+        extends ManuallyTriggeredScheduledExecutorService implements ComponentMainThreadExecutor {
+
+    private Thread executorThread = null;
+
+    @Override
+    public void assertRunningInMainThread() {
+        // always true
```

Comment:
I think this is not correct. You might have a component which uses another `ExecutorService` and wants to check that we are not accidentally running outside of the main thread executor.